lightning/src/util/ser_macros.rs

```diff
@@ -372,7 +372,87 @@ macro_rules! impl_writeable_tlv_based {
 				}, {
 					$(($vectype, $vecfield)),*
 				});
-				_init_tlv_based_struct!({$($reqfield),*}, {$($field),*}, {$($vecfield),*})
+				_init_tlv_based_struct!($st, {$($reqfield),*}, {$($field),*}, {$($vecfield),*})
+			}
+		}
+	}
+}
+
+/// Implement Readable and Writeable for an enum, with struct variants stored as TLVs and tuple
+/// variants stored directly.
+/// The format is, for example
+/// impl_writeable_tlv_based_enum!(EnumName,
+///   (0, VariantName) => { (0, variant_field)}, {(1, variant_optional_field)}, {};
```

Comment:
It wasn't quite intuitive to me what using this macro would look like with multiple struct fields, had to hunt for an example. Could you change this line to be:
```
 ///   (0, VariantName1) => { (0, variant_field)}, {(1, variant_optional_field)}, {},
 ///   (4, VariantName2) => {(0, variant_field_a), (1, variant_field_b)}, {}, {3, vec_field_c};